WPAndroid Integration.md

```diff
@@ -0,0 +1,102 @@
+### Summary
+
+WPAndroid by default integrates `react-native-bridge` as a binary dependency that's fetched from a remote maven repo. The CI in this repo will deploy a new version each time a commit is pushed to `develop` or an open PR, as well as each time a tag is created. This is the only required dependency. Whenever a new version is deployed, we can simply update `ext.gutenbergMobileVersion` in [build.gradle](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/build.gradle).
+
+During development, we can use composite builds and dependency substitution to point WPAndroid to our local checkout of `gutenberg-mobile`. To do this, we need to copy the [local-builds.gradle-example file](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/local-builds.gradle-example) in WPAndroid renaming it to `local-builds.gradle` and change the `localGutenbergMobilePath` value. Building the project as such will make WPAndroid build the source code for `react-native-bridge`, `react-native-aztec` and other sublibraries from the locally checked out folder. It'll also expect the metro server to be running to fetch the JS bundle.
+
+- [Work with local gutenberg-mobile checkout](#work-with-local-gutenberg-mobile-checkout)
+- [Deploy a new version of `react-native-bridge` from CI](#deploy-a-new-version-of-react-native-bridge-from-ci)
+- [Manually deploy a new version of `react-native-bridge`](#manually-deploy-a-new-version-of-react-native-bridge)
+- [Update WPAndroid's binary version](#update-wpandroids-binary-version)
+- [How to](#how-to)
+    - [How to test a gutenberg-mobile PR in WPAndroid](how-to-test-a-gutenberg-mobile-pr-in-wpandroid)
+    - [How to share WIP WPAndroid APK](how-to-share-wip-wpandroid-apk)
+    - [How to checkout gutenberg-mobile to deployed binary version](#how-to-checkout-gutenberg-mobile-to-deployed-binary-version)
+
+---
+
+### Work with local gutenberg-mobile checkout
+
+- Make sure you run `npm install` or `npm ci`
```

Comment:
Although I agree with you, the reason I wrote it this way is because `npm install` is the default suggestion everywhere in `gutenberg-mobile` and `gutenberg` repos, at least as far as I can tell. Going against that in this document didn't feel right to me.

I'll leave this decision to @mchowning.

Reply:
I would either leave it as-is or just have `npm install` since that is the default suggestion elsewhere.

> we're interested in consuming the local checkout more than modifying its libraries versions

Is this necessarily true? Won't some devs be using the local gutenberg-mobile checkouts for development?

Reply:
I am going to leave it as is. Since the CI does `npm ci`, I think it's good to have that as an alternative.

> Is this necessarily true? Won't some devs be using the local gutenberg-mobile checkouts for development?

Yes they would, that's the whole idea.

Reply:
Cool 👍 

> > Is this necessarily true? Won't some devs be using the local gutenberg-mobile checkouts for development?

> Yes they would, that's the whole idea.

Out of curiosity, would that entail much work around the Node modules?

Reply:
> Out of curiosity, would that entail much work around the Node modules?

Not the vast majority of the time, but it could.